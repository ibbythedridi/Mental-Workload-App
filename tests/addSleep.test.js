import React from 'react';
import renderer from 'react-test-renderer';
import AddSleep from '../screens/addSleep';


it('renders correctly', () => {
    const tree = renderer.create(<AddSleep />).toJSON();
    expect(tree).toMatchSnapshot();
});

// This checks the correct component(s) are existent in the JSON (views and buttons)
describe("<AddScreenTime />", () => {
  it('has 6 children', () => {
    const tree = renderer.create(<AddSleep />).toJSON();
    expect(tree.children.length).toBe(6);
  });
});