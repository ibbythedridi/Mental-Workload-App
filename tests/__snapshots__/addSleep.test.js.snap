// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Date: DD/MM/YYYY"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#eee",
        "borderRadius": 7,
        "elevation": 3,
        "height": 40,
        "marginBottom": 20,
        "padding": 5,
        "shadowColor": "#999",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Hours in Bed"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#eee",
        "borderRadius": 7,
        "elevation": 3,
        "height": 40,
        "marginBottom": 20,
        "padding": 5,
        "shadowColor": "#999",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Hours until Sleep"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#eee",
        "borderRadius": 7,
        "elevation": 3,
        "height": 40,
        "marginBottom": 20,
        "padding": 5,
        "shadowColor": "#999",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    onChangeText={[Function]}
    placeholder="Number of Times Woken up Through the Night"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#eee",
        "borderRadius": 7,
        "elevation": 3,
        "height": 40,
        "marginBottom": 20,
        "padding": 5,
        "shadowColor": "#999",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    onChangeText={[Function]}
    placeholder="Sleep Quality Rating (0-5)"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#eee",
        "borderRadius": 7,
        "elevation": 3,
        "height": 40,
        "marginBottom": 20,
        "padding": 5,
        "shadowColor": "#999",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#0043ff",
          "borderRadius": 7,
          "marginVertical": 5,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
